SingularityService/src/main/java/com/hubspot/singularity/auth/SingularityAuthorizationHelper.java

```diff
@@ -138,13 +143,17 @@ public void checkForAuthorization(SingularityRequest request, Optional<Singulari
     checkUnauthorized(user.isPresent(), "user must be present");
 
     final Set<String> userGroups = user.get().getGroups();
+    final Set<String> readWriteGroups = request.getReadWriteGroups().or(Collections.<String>emptySet());
+    if (request.getGroup().isPresent()) {
+      readWriteGroups.add(request.getGroup().get());
```

Comment:
My bad. I didn't see that 

I can't really use `Sets.union()` because `request.getGroup().get()` is just a String. I can just create a `new HashSet<>()` and add whatever `isPresent()` to it